File: wasm/sample/src/custom/mapping.ts
import { MappingList } from "../xplpc/data/mapping-list";
import { MappingItem } from "../xplpc/map/mapping-item";
import { Callback } from "./callback";

export class Mapping {
    public static initialize() {
        MappingList.shared().add(
            "platform.battery.level",
            new MappingItem(Callback.batteryLevel),
        );
    }
}

File: kotlin/android/sample/app/src/main/java/com/xplpc/runner/custom/Mapping.kt
package com.xplpc.runner.custom

import com.xplpc.data.MappingList
import com.xplpc.map.MappingItem

object Mapping {
    fun initialize() {
        MappingList.add("platform.battery.level", MappingItem(Callback::batteryLevel))
    }
}
